File: 231213/마라톤 중간에 택시타기 2/taking-a-taxi-in-the-middle-of-the-marathon-2.py
N = int(input())
pointList = [list(map(int, input().split())) for _ in range(N)]

def calcDistance(p1, p2):
    r1, c1 = p1
    r2, c2 = p2

    return abs(r1 - r2) + abs(c1 - c2)

import sys

minDis = sys.maxsize

for skip in range(1, N - 1):
    tempDistance = 0
    prevIdx = 0

    for i in range(1, N):
        if i == skip:
            continue

        tempDistance += calcDistance(pointList[prevIdx], pointList[i])
        prevIdx = i

    minDis = min(minDis, tempDistance)

print(minDis)

File: 231213/마라톤 중간에 택시타기 2/taking-a-taxi-in-the-middle-of-the-marathon-2.js
const fs = require('fs');
const inputAll = () => fs.readFileSync('/dev/stdin').toString().trim().split('\n');
const print = (...args) => console.log(...args);

const calcDistance = (p1, p2) => {
    const [r1, c1] = p1;
    const [r2, c2] = p2;

    return Math.abs(r1 - r2) + Math.abs(c1 - c2);
}

let [N, ...arr] = inputAll()

N = Number(N)
arr = arr.map(line => line.split(' ').map(elm => Number(elm)))

let minDis = Infinity;

for(let skip = 1; skip < arr.length - 1; skip++){
    let tempDis = 0;
    let prevIdx = 0;

    for(let i = 1; i < arr.length; i++){
        if(i === skip) continue;

        tempDis += calcDistance(arr[prevIdx], arr[i]);
        prevIdx = i;
    }

    minDis = Math.min(minDis, tempDis);
}

print(minDis)